src/Service/Results/SynonymRule.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace SilverStripe\Forager\Service\Results;
+
+use JsonSerializable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\View\ViewableData;
+
+class SynonymRule extends ViewableData implements JsonSerializable
+{
+
+    use Injectable;
+
+    public const TYPE_EQUIVALENT = 'TYPE_EQUIVALENT';
+    public const TYPE_DIRECTIONAL = 'TYPE_DIRECTIONAL';
+
+    /**
+     * Different services have different names for their synonym types
+     *
+     * Elasticsearch:
+     *  * TYPE_EQUIVALENT = Equivalent synonyms
+     *  * TYPE_DIRECTIONAL = Explicit synonyms
+     *
+     * Algolia:
+     * * TYPE_EQUIVALENT = Regular synonyms
+     * * TYPE_DIRECTIONAL = One-way synonyms
+     *
+     * Typesense:
+     * * TYPE_EQUIVALENT = Multi-way synonyms
+     * * TYPE_DIRECTIONAL = One-way synonyms
+     */
+    private ?string $type = null;
+
+    /**
+     * Used for "directional" synonyms
+     *
+     * Different services call this value (or values) different things
+     *
+     * * Elasticsearch: "Left hand side" of the arrow ("=>") declaration
+     * * Algolia: Input
+     * * Typesense: Root
+     *
+     * All have the same result though. This value (or values) will be transformed into your synonym values
+     *
+     * @var string[]
+     */
+    private array $root = [];
+
+    /**
+     * @var string[]
+     */
+    private array $synonyms = [];
+
+    public function __construct(private readonly string|int $id)
+    {
+        parent::__construct();
+    }
+
+    public function getID(): int|string|null
+    {
+        return $this->id;
+    }
+
+    public function getType(): ?string
+    {
+        return $this->type;
+    }
+
+    public function setType(?string $type): static
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    public function getRoot(): array
+    {
+        return $this->root;
+    }
+
+    /**
+     * Useful method to use if you want to display this data in (for example) a GridField
+     */
+    public function getRootAsString(): string
+    {
+        return implode(', ', $this->root);
+    }
+
+    public function addRoot(string $value): static
+    {
+        $this->root[] = $value;
+
+        return $this;
+    }
+
+    public function setRoot(array $values): static
+    {
+        # Performing a loop on the array instead of direct assignment effectively checks that all values are of the
+        # type/s expected by addRoot()
+        foreach ($values as $value) {
```

Comment:
this won't remove existing values as the name suggests

Reply:
Very true. Good spot